presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlTypeMapping.java

```diff
@@ -291,11 +309,39 @@ private void testUnsupportedDataType(String databaseDataType)
                 Function.identity());
     }
 
+    public static DataType<String> jsonDataType()
+    {
+        return DataType.dataType(
+                "json",
+                JSON,
+                value -> format("JSON'%s'", value),
+                Function.identity());
+    }
+
+    @Test
+    public void jsonDataTypeTest()
+    {
+        DataTypeTest dataTypeTest = DataTypeTest.create()
+                .addRoundTrip(jsonDataType(), "{}")
+                .addRoundTrip(jsonDataType(), "{\"a\":1,\"b\":2}")
+                .addRoundTrip(jsonDataType(), "{\"a\":[1,2,3],\"b\":{\"aa\":11,\"bb\":[{\"a\":1,\"b\":2},{\"a\":0}]}}")
+                .addRoundTrip(jsonDataType(), "[]");
+
+        dataTypeTest.execute(getQueryRunner(), prestoCreateAsSelect("presto_test_json"));
+        dataTypeTest.execute(getQueryRunner(), postgresCreateAsSelect("tpch.postgresql_test_json1"));
+        dataTypeTest.execute(getQueryRunner(), postgresCreateAndInsert("tpch.postgresql_test_json2"));
+    }
+
     private DataSetup prestoCreateAsSelect(String tableNamePrefix)
     {
         return new CreateAsSelectDataSetup(new PrestoSqlExecutor(getQueryRunner()), tableNamePrefix);
     }
 
+    private DataSetup postgresCreateAsSelect(String tableNamePrefix)
```

Comment:
Why do you think this is needed? We already have `postgresCreateAndInsert`.
We're not testing Postgres here, so it should not matter whether we use CTAS or INSERTs to setup the data in Postgres.

Reply:
make sense. done